### ISSUE TITLE
Identity not found issue on AKS running on MSI

### ISSUE THREAD

**Issue by vishnujanardhanan** [Azure/aad-pod-identity#987]
**Have you**

- [ ] Read [Troubleshooting Guide](../../docs/readmes/README.troubleshooting.md)
- [ ] Read [Known Issues](../../docs/readmes/README.known-issues.md)
- [ ] Searched on [GitHub issues](https://github.com/Azure/aad-pod-identity/issues)

We seem to get Identity not found error on AKS cluster spun up using MSI. This is causing issues with az login --identity and while pulling images from acr. It works for sometime and starts to fail

**Steps To Reproduce**

**Expected behavior** Identity to work as normal

**AAD Pod Identity version** - 1.17

**Kubernetes version** 1.18.14

**Additional context** - NMI, MIC pod logs available on azure support ticket 2103040050001463

**Comment by aramase** [comment]
@vishnujanardhanan I looked at the support ticket. MIC uses the underlying cluster identity to authenticate and make cloud API calls to Azure for assigning/unassigning the identity. It looks like the issue here is the cluster identity is in fact not found. If you assign the agent pool identity to the VMSS it should work fine.

**Comment by aramase** [comment]
@vishnujanardhanan I looked at the internal logs. The AKS kubelet identity was deleted by pod identity component MIC. This happens when the kubelet identity is configured for use by application using `AzureIdentity`. The way pod identity component MIC works is by assigning the identity defined in the `AzureIdentity` when 1st pod using it is created and un-assigns the identity when the last pod using it is deleted. This means when all pods using the kubelet identity were deleted, it got un-assigned by MIC. MIC however relies on the kubelet identity to communicate with azure. If using the kubelet identity for application pod, it is recommended to configure the kubelet identity as immutable identity using https://azure.github.io/aad-pod-identity/docs/configure/feature_flags/#immutableusermsis-flag. By setting the kubelet identity as immutable, MIC will skip unassigning it.

We have already merged a PR to skip un-assigning kubelet identity if used by the application (configured in `AzureIdentity`): https://github.com/Azure/aad-pod-identity/pull/981. This will be available in the next release v1.7.5.

Mitigation steps:
1. Manually assign the kubelet identity by running "az vmss identity assign -g <resource group> -n <vmss name> --identities /subscriptions/<sub id>/resourcegroups/MC_mgmt-euw-v2_mgmt-euw-v2_westeurope/providers/Microsoft.ManagedIdentity/userAssignedIdentities/mgmt-euw-v2-agentpool"
2. Add the immutable identities flag in MIC with the kubelet identity configured: "--immutable-user-msis=<kubelet identity clientID>"

These steps will not be necessary when we have a new release but configuring kubelet identity as immutable identity is required to ensure the identity doesn't get removed.
